types/capnp.go

```diff
@@ -0,0 +1,26 @@
+package types
+
+import "fmt"
+
+func CoinFromCoinE(e CoinE) (coin Coin, err error) {
```

Comment:
I like this approach.

We can 
(1) define powerful, deterministic zero-copy serialization format with cap'n proto and
(2) define functions to convert these cap'n proto structs to any custom go struct we want (or any custom js struct, etc)

The overhead should be comparable to parsing the data with another codec and we can use language-specific idioms.

Reply:
The downside is that this requires a conversion function to be written for each struct that we want to use as a normal struct in code. This is not too much work, but is boilerplate that devs abhor.

This function could have been created by codegen and maybe enhancing that codegen would make this super-smooth.

We could use cap'n proto for serialization and canonical signing. Then write custom code to parse it into convenient structs to work with. Once we are happy with these hand-crafted conversion functions and the only complaint is the hand-coding of them, we could automate them with codegen - but only when we are all clear on what we want.